builtins: implement ST_Force2D

otan opened the pull request:
This unblocks use of GeoServer.

Force_3D, 4D, etc. variants are not yet implemented as we do not store
3D/4D objects in CRDB yet. But the building blocks are all there.

Resolves https://github.com/cockroachdb/cockroach/issues/48933.

Release note (sql change): Implement the ST_Force2D functionality for
geometry types.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/51514)
<!-- Reviewable:end -->

rytaft reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/> 

> Reviewed 5 of 5 files at r1.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51514#-:-MCS8aBZ2Di1jbKytODJ:bg7igkv)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @otan)

___
*[pkg/geo/geomfn/force_test.go, line 92 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51514#-MCS792k4qsy2LL2WPsy:-MCS792l4nExF4mgXlKV:b-v41n19) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e922e8b00ca37b2e40c313a4d99143b728c2c2a/pkg/geo/geomfn/force_test.go#L92)):*
> ```Go
> 			).SetSRID(4326),
> 			geom.XYZM,
> 			geom.NewPolygonFlat(
> ```

how about adding tests for some or all of XYZM -> XYZ, XYZM -> XYM, XYZM -> XY


<!-- Sent from Reviewable.io -->

otan committed:
builtins: implement ST_Force2D

This unblocks use of GeoServer.

Force_3D, 4D, etc. variants are not yet implemented as we do not store
3D/4D objects in CRDB yet. But the building blocks are all there.

Release note (sql change): Implement the ST_Force2D functionality for
geometry types.

otan reviewed (commented):
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51514)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @rytaft)

___
*[pkg/geo/geomfn/force_test.go, line 92 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/51514#-MCS792k4qsy2LL2WPsy:-MCSJ40w9p7twyrp56RZ:b-896fix) ([raw file](https://github.com/cockroachdb/cockroach/blob/4e922e8b00ca37b2e40c313a4d99143b728c2c2a/pkg/geo/geomfn/force_test.go#L92)):*
<details><summary><i>Previously, rytaft (Rebecca Taft) wrote…</i></summary><blockquote>

how about adding tests for some or all of XYZM -> XYZ, XYZM -> XYM, XYZM -> XY
</blockquote></details>

Done.


<!-- Sent from Reviewable.io -->

rytaft reviewed (approved):
> Reviewed 1 of 1 files at r2.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/51514)* status: :shipit: complete! 0 of 0 LGTMs obtained (and 1 stale)



<!-- Sent from Reviewable.io -->

otan commented:
bors r=rytaft

otan commented:
bors r=rytaft

otan commented:
bors r=rytaft

bors not happy today
